off-chain/src/Network/Types.purs

```diff
@@ -89,23 +158,46 @@ newtype LeaderState a = LeaderState
   , numberOfActionsRequestsMade :: MutableInt
   }
 
+addAction :: forall a. UserAction a -> LeaderState a -> Aff UUID
+addAction action st = liftEffect do
+  actionUUID <- genUUID
+  pushRefMap_ actionUUID action
+    (unwrap st).pendingActionsRequest
+  pure actionUUID
+
+numberOfPending :: forall a. LeaderState a -> Aff Int
+numberOfPending st = OMap.length <$>
+  (liftEffect $ Ref.read (unwrap st).pendingActionsRequest)
+
 derive instance Newtype (LeaderState a) _
 
+newtype LeaderConfiguration :: forall k. k -> Type
 newtype LeaderConfiguration a = LeaderConfiguration
   { maxWaitingTimeForSignature :: MiliSeconds
   , maxQueueSize :: Int
   , numberOfActionToTriggerChainBuilder :: Int
   , maxWaitingTimeBeforeBuildChain :: Int
-  , serverHandlers ::
-      { includeAction :: UserAction a -> Aff $ Either IncludeActionError UUID
-      , acceptSignedTransaction ::
-          SendSignedTransaction
-          -> Aff $ Either AcceptSignedTransactionError Unit
-      , rejectToSign :: UUID -> Aff Unit
-      , getActionStatus :: UUID -> Aff StatusResponse
-      }
+
+  -- FIXME: not sure we should do it like this.
```

Comment:
I think all this handlers will be actually `payload`'s handlers, that will user interface functions for `LeaderNode`